### PR TITLE
Fixing Mailchimp not saving name. Refactor reCaptcha file structure.

### PR DESCRIPTION
**Pull request by tonykieling**
|                                    Web Dev Path                                    |
| :--------------------------------------------------------------------------------: |
| [**187**](https://github.com/MarianaSouza/web-dev-path/issues/187) |

#### Have you updated the CHANGELOG.md file? If not, please do it.
Yes.

#### What is this change?
- Contact and Subscribe forms are saving Name on Mailchimp.
- Changed /api to have a file to check reCaptcha and another to send email.
- Refactored <NewsletterSubscribe> and <ContactUsForm> to handle Mailchimp and GoogleRecaptcha in one place (<Mailchimp>), and utilize the apis above mentioned.

#### Were there any complications while making this change?
The initial settings and get used to Mailchimp.

#### How did you verify this change?
Checked on Chrome, FF, and MS Edge.

#### When should this be merged?
After review.

### CODE REVIEW

File: temp-notes.txt
Comment (cherylli) [inline temp-notes.txt:1]: @tonykieling you left a "temporary" file here but it might be good to include somewhere maybe in the wiki for future collborator

Comment (tonykieling) [reply temp-notes.txt:1]: Thanks @cherylli for pointing that out - I forgot to delete the temp file.
New commit with file deleted already in place.

**Review by mariana-caldas — APPROVED** [review]
That's awesome work, @tonykieling ! 🥇 

Thanks for not only solving the problem but also refactoring the MailChimp and reCaptcha structures by separating the logic while adding more error messages. 🥷

File: pages/api/sendEmail.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export default async (email, name, subject, message, subscribe) => {
+  try {
+    // receiverEmail: The email will be sent here
+      const receiverEmail = 'hello@webdevpath.co';
+    // sendgridEmail: This is the email verfied by sendgrid
+    // the email will appear to be sent from this email
+    // If a non verified email is used, we get a 403 error
+      const sendgridEmail = 'hello@webdevpath.co';
+
+    const emailContent = `
```
[the remark this answers is not in the document]
Comment (tonykieling) [reply pages/api/sendEmail.js:15]: Thanks for the feedback @mariana-caldas .

**Comment by Satoshi-Sh** [comment]
I would like to test this out on my end but I'm not sure how to set up MailChimp. I just signed up. If you can guide me, it would be great. 

Do I need to get my own recaptcha and SendGrid keys as well? I'm not familiar with them either.

**Comment by mariana-caldas** [comment]
> I would like to test this out on my end but I'm not sure how to set up MailChimp. I just signed up. If you can guide me, it would be great.
> 
> Do I need to get my own recaptcha and SendGrid keys as well? I'm not familiar with them either.

Hey @Satoshi-Sh !

You can follow these approaches to test it:

* _Recommend approach_: By using an email you have access to, even the same you have already subscribed to in this format `emailsubscribed+1oranyting@gmail.com`, you try to subscribe to our Newsletter through both our component (the Newsletter section we see on every page but Contact) and the Contact form. If you see the expected successful messages, it means MailChimp is working as intended. I will only have to remove those "fake" test contacts from the Mailchimp list later.

* By creating a Mailchimp account and list and changing the keys to your to see if contacts go to that list properly. This approach is the best one if you are this feature developer, but I don't think it's necessary...

In both cases, you don't need to create other Recaptcha or SendGrid keys.

Your question is interesting, Satoshi, and makes me wonder if we should add a section to the PR template or an explanation to our [wikipedia](https://github.com/Web-Dev-Path/web-dev-path/wiki/Creating-a-Pull-Request), encouraging developers to describe how to test the features/fixes. What do you think, @cherylli  and @tonykieling ?

**Comment by cherylli** [comment]
Wiki sounds like a good idea

**Comment by Satoshi-Sh** [comment]
The updated code looks nice. 

I tested the two components. Let me know if you can see my name, @mariana-caldas . After that, I will approve this pull request.

**Comment by mariana-caldas** [comment]
> The updated code looks nice.
> 
> I tested the two components. Let me know if you can see my name, @mariana-caldas . After that, I will approve this pull request.

You were subscribed correctly, @Satoshi-Sh ! I've also received your email from the contact form 🙌🏼

**Review by Satoshi-Sh — APPROVED** [review]
Thanks for the detailed proecess documentation!!

**Comment by tonykieling** [comment]
If we decide to have this info on the wiki, I could elaborate a step-by-step to set Mailchimp, Sendgrid and GoogleReCaptcha (.env) for new volunteers.

**Comment by mariana-caldas** [comment]
> If we decide to have this info on the wiki, I could elaborate a step-by-step to set Mailchimp, Sendgrid and GoogleReCaptcha (.env) for new volunteers.

Thanks, @tonykieling !
I think we only need to add to our [wiki](https://github.com/Web-Dev-Path/web-dev-path/wiki/Creating-a-Pull-Request) a section explaining that depending on the feature, the developer should include how to test the feature under the PR description item _"How did you verify the change"_ Thoughts, @cherylli ?

**Comment by cherylli** [comment]
> > If we decide to have this info on the wiki, I could elaborate a step-by-step to set Mailchimp, Sendgrid and GoogleReCaptcha (.env) for new volunteers.
> 
> Thanks, @tonykieling ! I think we only need to add to our [wiki](https://github.com/Web-Dev-Path/web-dev-path/wiki/Creating-a-Pull-Request) a section explaining that depending on the feature, the developer should include how to test the feature under the PR description item _"How did you verify the change"_ Thoughts, @cherylli ?

I was more thinking like a brief explanation of how these things (mailchimp, recaptcha, etc) work together as it might be confusing for newer developers, since Tony already wrote something about it (https://github.com/Web-Dev-Path/web-dev-path/blob/06a0bc22469fa7b7a88abc71fc7a4c951e79ea9a/temp-notes.txt), I thought it might be for other future developers

**Comment by mariana-caldas** [comment]
> > > If we decide to have this info on the wiki, I could elaborate a step-by-step to set Mailchimp, Sendgrid and GoogleReCaptcha (.env) for new volunteers.
> > 
> > 
> > Thanks, @tonykieling ! I think we only need to add to our [wiki](https://github.com/Web-Dev-Path/web-dev-path/wiki/Creating-a-Pull-Request) a section explaining that depending on the feature, the developer should include how to test the feature under the PR description item _"How did you verify the change"_ Thoughts, @cherylli ?
> 
> I was more thinking like a brief explanation of how these things (mailchimp, recaptcha, etc) work together as it might be confusing for newer developers, since Tony already wrote something about it (https://github.com/Web-Dev-Path/web-dev-path/blob/06a0bc22469fa7b7a88abc71fc7a4c951e79ea9a/temp-notes.txt), I thought it might be for other future developers

I'd say it's better to redirect new developers to the official documentation of these platforms. New members need to know they have to get the .env variables because we use those API platforms, that's completely enough. Thanks, @cherylli and @tonykieling !
By the way, @cherylli , can we get your PR approval so Tony can merge this PR? Thanks!

**Comment by cherylli** [comment]
> 

Sorry, i've been really busy lately, I haven't had a chance to pull and setup all my mailchimp accounts etc to test this PR, and I believe my sendgrid account is banned or something. I might be able to in the weekend but it's fathers day here so i'll probably be gone all day, or would you like me to approve without checking or running it locally?

**Comment by mariana-caldas** [comment]
> 

Hey @cherylli !
To test the feature, you can use the "recommended approach" I explained above to Satoshi. You don't need to set anything. You only need to run the PR locally and try to send a request to subscribe both on the contact page and the Newsletter section. I would check our MailChimp list to validate your "tests" to remove them later. 
I can always merge the PR manually, so don't sweat if can't, Cheryl! 

That's a great opportunity to learn from a great PR, @LuSilvaDeveloper , @sergiobtos , and @bernardoasc.

**Comment by cherylli** [comment]
> > 
> 
> Hey @cherylli ! To test the feature, you can use the "recommended approach" I explained above to Satoshi. You don't need to set anything. You only need to run the PR locally and try to send a request to subscribe both on the contact page and the Newsletter section. I would check our MailChimp list to validate your "tests" to remove them later. I can always merge the PR manually, so don't sweat if you can't, Cheryl! Btw, that's why I reinforce the importance of adding something to our wiki (create PR) so developers detail how to test things when necessary.
> 
> That's a great opportunity to learn from a great PR, @LuSilvaDeveloper , @sergiobtos , and @bernardoasc.

ok sure I can do that sometime in the weekend, I just prefer doing the full testing myself

**Pull request merged**